lib/counter_culture/with_connection.rb

```diff
@@ -0,0 +1,19 @@
+module CounterCulture
+  class WithConnection
+    def initialize(recipient)
+      @recipient = recipient
+    end
+
+    attr_reader :recipient
+
+    def call
+      if recipient.respond_to?(:with_connection)
+        recipient.with_connection do |connection|
+          yield connection
+        end
+      else
+        yield recipient.connection
+      end
```

Comment:
```suggestion
      if Gem::Version.new(Rails.version) < Gem::Version.new('7.2.0')
        yield recipient.connection
      else
        recipient.with_connection do |connection|
          yield connection
        end
      end
```

How do you feel about using the Rails version here instead to determine which method to call? In my head, if we're in Rails 7.2 or later, where `with_connection` should be available, it would be an error if `respond_to?(:with_connection)` was false. I realize I'm nitpicking and this probably doesn't matter in reality, but I _think_ I have a preference for using `Rails.version`. Do let me know if I'm missing something there though!

Reply:
I don't have a strong preference, so happy to switch to a Rails version check. One tiny change though, is `with_connection` is available from 7.1; 7.2 just adds the option to deprecate/disable the old API. So I would go with checking against 7.1 if you don't have any concerns? As a side note, we're already running this in a fairly big 7.1 production application, and haven't noticed any problems so far.

Reply:
Yeah, let's switch it then. If it was added in 7.1, great, let's use that! The specs would let us know if it wasn't available. 😄

Reply:
The specs did catch something: apparently, in 7.1 `with_connection` is not available on `ActiveRecord::Base` class and its instance, but it available on `ConnectionPool`, so I've added a bit more logic to handle that.